core/modules/wdb/QueryRunner.h

```diff
@@ -46,10 +46,21 @@
 
 namespace lsst {
 namespace qserv {
+
 namespace proto {
 class ProtoHeader;
 class Result;
-}}}
+}
+
+namespace util {
+class TimerHistogram;
```

Comment:
Forward declarations need to be indented.

Reply:
Where is this stated? They aren't indented anywhere else (aside from replica) and "The contents of namespaces are not indented." (6-21b).

Reply:
CC: @n8pease @andy-slac @fritzm 

I was "beaten to death" by members of our team when I expressed a similar objection in one of my tickets's review 😸 Claims I heard included:
"...please, do what the rest of Qserv does..."
"...it's a common practice do indent forward declaration..."
In the end, I just gave up and did what they wanted from me. So, I'm just 'forwarding" my pain to you 😸 Feel free to ignore my suggestion though.
And, speaking of which code does it one way or another, the `replica` module is a big contributor, but not the only one. I've counted a number of headers where indents (the first case are used) versus the ones where they're aren't:
```
% egrep '   class .+;' core/modules/*/*.h | awk -F: '{print $1}' | sort -u | wc -l
91
% egrep '^class .+;' core/modules/*/*.h | awk -F: '{print $1}' | sort -u | wc -l
51
```
An this is the frequency (the number of headers) in each module where the indents are used in the forward declarations:
```
 1 ccontrol
 1 global
 1 mysql
 2 parser
 7 qana
 1 qdisp
 3 qproc
30 query
31 replica
 1 rproc
 1 sql
 1 util
 3 wbase
 1 wcontrol
 3 wdb
 2 wpublish
 2 wsched
```
And there is a list of counters for modules where no spacing is present in front of the forward declarations:
```
7 ccontrol
2 css
1 czar
2 memman
3 mysql
1 parser
1 proto
2 qana
6 qdisp
1 qhttp
2 qproc
2 query
3 replica
1 util
4 wbase
2 wdb
7 wpublish
1 wsched
3 xrdsvc
```

Reply:
Not only my heart is with you, but the LSST C++ Style guide itself says:
"...The contents of namespaces are not indented..."

See:
https://developer.lsst.io/cpp/style.html#b-nested-namespaces-should-be-aligned-left-with-each-level-of-nesting-on-a-new-line

I wish I had read that before adding indents in my forward declarations 😸

Reply:
Igor, your `grep` may be finding class declarations inside classes, which should be indented.

Reply:
That's certainly true. I have inspected the headers, and found 12 "false-positives". Still, the majority of forward declarations are indented.

I'm not arguing that the code had to be fixed (one way or another). I think this is one of the topics we should never touch upon on the code reviews. I myself was quite frustrated when I was requested to add those indents because "...the rest of Qserv code does it...". I think we (always) have a bigger "fish to fry" 😸 

Here is my proposal of *DONT'S* for the code reviews (this list could be extended):
- spacing, indents (except the standard ones for code blocks)
- alignments 
- empty lines
- the line length (exceeding 100 or whatever number of characters) should  be fine as long as it's needed
- code documentation (it has to be up to a developer how or if to document their code, using Doxygen, etc.).

These are the ones which are causing most disagreements and confusions. If people feel we really need to unify the "look & feel" of our code then one of those automated code formatting tools would help.

Reply:
^so basically, the complete list of usual Igor sins ;-)  I'd schedule us some group therapy during PCW, but I guess that's what drinks by the pool are for 🍺

Reply:
More seriously, after taking a closer look, @jgates108 and @iagaponenko  I think the "prevailing style" could be described as:

- For "short" namespaces (in particular, those near the head of a file containing a few forward declarations), **DO** indent, because indenting those improves readability.
- For "long" namespaces (in particular, those spanning the entirety or a class declaration or method definitions, "set and forget" for majority of a file) **DO NOT** indent, because indenting those does not improve readability.

What do you think?

Reply:
Honestly, indent sometimes sounds awful to me. Difference between short and long is? You have a short one and a long one together?
I think Forward declarations are not at all interesting. They just need to be there. I'd rather not have any special rules for them.

Reply:
To be clear, I have nothing against Igor (or anyone) going through and marking style issues. If there is an existing style rule, I'll follow it.  If there isn't, I can ignore him :)

If there is not an existing rule, I don't want to make more unless REALLY necessary. If Igor really wants to indent his forward declarations, it doesn't really bother me.

Reply:
For clarity only: the difference between "long" and "short" to me is whether you parse it visually as single chunk or not.  Anything that involves scrolling around certainly does not qualify.

However, one person's readable is another person's fugly (c.f. the recent discussion about Doxygen comments).  We've certainly bigger fish, so I am offering an opinion but no argument.

Reply:
As per the classic: "Indent or not indent - that is THE question!" 😸 

Seriously speaking, my only problem in this context is that all IDEs I've used so far always try to indent within namespaces whenever I'm adding a new class, function, variable into a namespace. Otherwise, either way works for me just fine. I have no philosophical reservations for this other than practical ones.

Reply:
And let's not start IDE holy war ⚔️